text/0000-release-channels.md

```diff
@@ -0,0 +1,70 @@
+
+We’ll likely need to update `bundle env` to include channel information.
+
+# Drawbacks
+
+We might not want to do this if we think it would be more development effort than is worth the benefits of having multiple release channels.
+
+This change increases the possible number of Bundler versions in the wild, so it will likely increase the workload for maintainers by at least a small amount.
+
+Adding a stable channel means supporting one older version of Bundler for longer than our current support plans, which is likely small but still some more additional work.
+
+# Rationale
+
+The Bundler team deals with the tension between stability and change on a daily basis. With the tens or hundreds of thousands of users that Bundler has, it feels impossible to satisfy everyone with every release.
+
+Release channels tries to make it possible to satisfy everyone, by dividing users into groups based on risk tolerance, and deliberately aiming for that level of risk. Adding stable releases might make it possible to support default releases for less time overall, to try to balance out the overall support load.
+
+The strategy proposed in this RFC (or something very similar) is currently in active use by the Rust language, the NPM package manager, and the Ember.js framework, among many others. It seems to strike the best balance, allowing ongoing development work while protecting production systems from changes and accidents as much as possible.
+
+If we don’t do this, we will continue to struggle with the existing problem of strongly conflicting expectations and tensions within the user base for each version of Bundler that is released.
+
+# Alternatives
+
+Instead of releasing to canary with every commit, we could release to canary on an automated cadence, like every 24 hours.
+
+Other possible solutions to this class of problem include: the Unix and early Ruby strategy of alternating stable and unstable releases, declaring all releases unstable except one blessed release per year or quarter, or any other option to classify some versions as stable and some as unstable.
+
+# Unresolved questions
+
+The details of how to release, when to release, how feature flags interact with release channels, and how to distinguish channel releases all likely need to be worked out before implementation.
+
+It would be a wonderful next step (albeit out of scope for this RFC) to ship and support one Heroku buildpack for each Bundler release channel, so that Ruby developers on Heroku can easily test each channel, give feedback, and provide bug reports _before_ Bundler releases are rolled out to all Heroku users.
+
+Questions about breaking changes, how long to support versions, support for versions of Ruby or RubyGems, and integration with RubyGems all seem out of scope for this specific RFC.
```

Comment:
might want to say about the subject of an LTS version, then?

Reply:
Sorry, I'm not sure what you mean by this. Can you elaborate or rephrase?

Reply:
Is the subject of a Bundler long term support version on the table for the RFC, or is another topic

Reply:
A true LTS version would be another RFC. This RFC is purely about providing opt-in release candences, and inherits the existing support policy, which is iirc “majors get features and bugfixes for one year, and might get bugfixes for a second year but backports are not guaranteed”.